magento/magento2
# [WIP] Fix import images with same filename in different directories

### Description
When you're trying to import one product with 2 images and both images haves the same filename but different directory only one image will be imported. The solution I proposed in the commit is not the-best-case solution, I need Magento2 architect feedback to complete this PR. Any help would be welcome.

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/7412

### Manual testing scenarios
See the CSV in attachment as example. [products.csv.zip](https://github.com/magento/magento2/files/1321516/products.csv.zip)

I'll try to explain with an example: if you want to import some product with 2 images. You have one image in directory /images/dir1/food.jpg and you have a second image in /images/dir2/food.jpg only the first image will be imported. The reason for that is the files while importing not are renamed. The check to see if that file already exists was default set to true but must be false.

### Problem
If you run the import twice or more it will create a new image for every image (adding _2, _3, _4, ...). It will keep adding images to the catalog/product directories to keep them unique. This is not best case if you run imports multiple times. I don't see the exact solution right now, any help would be welcome.

Thanks in advance.

### Contribution checklist
